### PR TITLE
fix(fmt): stable formatting of HTML files with JS

### LINKED ISSUE #27030: fmt(svelte): error with types in snippet
Version: Deno 2.1.1

```svelte
{#snippet snip(num: number)}
  <div>Test {num}</div>
{/snippet}
```

```txt
deno fmt --unstable-component
Error formatting: C:\Temp\deno-init\test.svelte
   Expected ';', '}' or <eof> at file:///C:/Temp/deno-init/test.svelte.tsx:1:29

         <>{snip(num: number)}</>
                              ~
Checked 4 files
```

without the `: number ` it works fine.

### PR DESCRIPTION
**Pull request by bartlomieju**
Closes https://github.com/denoland/deno/issues/26407
Closes https://github.com/denoland/deno/issues/26763
Closes https://github.com/denoland/deno/issues/26560
Closes https://github.com/denoland/deno/issues/26744
Closes #27030

### CODE REVIEW

**Comment by bartlomieju** [comment]
This still is broken. 

For all three reproductions I get `indent_level` of 0 coming from `markup_fmt`.

**Comment by g-plane** [comment]
markup_fmt v0.18.0 is released.

Notes:

- It also fixes #27030 .
- New option `script_formatter` is introduced. For Deno, its value should be `Some(ScriptFormatter::Dprint)`.

**Comment by bartlomieju** [comment]
Thanks! Will try it out tonight.

File: cli/tools/fmt.rs [hunk 1]
```diff
@@ -549,11 +551,7 @@ pub fn format_sql(
   // Add single new line to the end of file.
   formatted_str.push('\n');
 
-  Ok(if formatted_str == file_text {
-    None
-  } else {
-    Some(formatted_str)
-  })
+  Ok(Some(formatted_str))
```
Comment (dsherret) [inline cli/tools/fmt.rs:554]: I think this should be reverted? Was there no test that fails?

Comment (dsherret) [reply cli/tools/fmt.rs:554]: https://github.com/denoland/deno/pull/27014/files#diff-fe46255b8864b7c270c3f211fd444d65e0290b72bdf1bc5aff9c1198c3576ff2

**Review by marvinhagemeister — APPROVED** [review]

**Review by dsherret — CHANGES_REQUESTED** [review]
Going to block this one until https://github.com/denoland/deno/pull/27187 is resolved

**Review by dsherret — APPROVED** [review]
LGTM

**Pull request merged**